sys/include/net/gnrc/ipv6/nib.h

```diff
@@ -0,0 +1,541 @@
+ * Also note, that on a 6LoWPAN router (and border router) the neighbor cache
+ * [does not behave like a cache](https://tools.ietf.org/html/rfc6775#page-11)
+ * but a registry, so the caching algorithm does not apply here either.
+ *
+ * @{
+ *
+ * @file
+ * @brief   NIB definitions
+ *
+ * @author  Martine Lenders <m.lenders@fu-berlin.de>
+ */
+#ifndef GNRC_IPV6_NIB_H_
+#define GNRC_IPV6_NIB_H_
+
+#include <stdint.h>
+
+#include "net/ipv6/addr.h"
+#include "net/ipv6/hdr.h"
+#include "net/gnrc/netif/hdr.h"
+#include "net/ndp.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef GNRC_IPV6_NIB_SIZE
+#if !defined(MODULE_GNRC_IPV6_ROUTER) || defined(DOXYGEN)
+#define GNRC_IPV6_NIB_SIZE  (32U)   /**< size of the NIB. */
+#else
+#define GNRC_IPV6_NIB_SIZE  (2U)    /* even 1 just for default route should
+                                     * suffice, but RFC 4861 suggests at least 2
+                                     * (https://tools.ietf.org/html/rfc4861#section-5.3) */
+#endif
+
+/**
+ * @brief   States for neighbor unreachability detection
+ *
+ * @see [RFC 4861, section 7.3.2](https://tools.ietf.org/html/rfc4861#section-7.3.2)
+ * @see [RFC 7048](https://tools.ietf.org/html/rfc7048)
+ */
+typedef enum {
+    GNRC_IPV6_NIB_NUD_STATE_UNMANAGED = 0,  /**< not managed by NUD */
+    GNRC_IPV6_NIB_NUD_STATE_UNREACHABLE,    /**< entry is not reachable */
+    GNRC_IPV6_NIB_NUD_STATE_INCOMPLETE,     /**< address resolution is currently performed */
+    GNRC_IPV6_NIB_NUD_STATE_STALE,          /**< address might not be reachable */
+    GNRC_IPV6_NIB_NUD_STATE_DELAY,          /**< NUD will be performed in a moment */
+    GNRC_IPV6_NIB_NUD_STATE_PROBE,          /**< NUD is performed */
+    GNRC_IPV6_NIB_NUD_STATE_REACHABLE,      /**< entry is reachable */
+} gnrc_ipv6_nib_nud_state_t;
+
+/**
+ * @brief States for 6LoWPAN address registration (6Lo-AR)
+ */
+typedef enum {
+    GNRC_IPV6_NIB_AR_STATE_NONE = 0,        /**< not managed by 6Lo-AR */
+    GNRC_IPV6_NIB_AR_STATE_GC,              /**< address can be removed when low on memory */
+    GNRC_IPV6_NIB_AR_STATE_TENTATIVE,       /**< address registration still pending */
+    GNRC_IPV6_NIB_AR_STATE_REGISTERED,      /**< address is registered */
+} gnrc_ipv6_nib_ar_state_t;
+
+/**
+ * @brief   A public NIB entry
+ */
+typedef struct {
+    const ipv6_addr_t dst;          /**< A destination or prefix to destination */
+    const ipv6_addr_t next_hop;     /**< Next hop to gnrc_ipv6_nib_t::dst */
+#if defined(MODULE_GNRC_SIXLOWPAN_ND_ROUTER) || defined(DOXYGEN)
+    /**
+     * @brief   The neighbors EUI-64 (used for DAD)
+     *
+     * 
+     */
+    eui64_t eui64;
+#endif
+    const uint8_t l2addr[GNRC_NETIF_HDR_L2ADDR_MAX_LEN];
+                                    /**< L2 address of next hop */
+    const unsigned pfx_len : 8;     /**< prefix-length of gnrc_ipv6_nib_t::dst */
+    const unsigned l2addr_len : 4;  /**< length of gnrc_ipv6_nib_t::l2addr */
```

Comment:
Does anyone remember why we not use this style of defining bytefields anymore? I find them really convenient.